jaeger-core/src/main/java/io/jaegertracing/Configuration.java

```diff
@@ -190,11 +190,20 @@ public static Configuration fromEnv() {
   }
 
   public static Configuration fromEnv(String serviceName) {
-    return new Configuration(serviceName)
-            .withTracerTags(tracerTagsFromEnv())
-            .withReporter(ReporterConfiguration.fromEnv())
-            .withSampler(SamplerConfiguration.fromEnv())
-            .withCodec(CodecConfiguration.fromEnv());
+    Configuration config = new Configuration(serviceName);
+    config.initFromEnv();
+    return config;
+  }
+
+  protected void initFromEnv() {
```

Comment:
why? this is not the code we want to repeat, so why is it factored out into a method?

Reply:
This is because `Configuration` has no factory method at the moment. All these static methods are problematic because they construct the concrete type instead of the subclass. In general, I was able to solve by passing or overriding the factory. I can try doing the same here.

Reply:
just an idea - while subclassing tracer / span is necessary, I don't think the same holds for the configuration class. It's fine to wrap it, and not provide all the functionality that exists in the oss. For example, internally we don't init from env.

Reply:
Agreed. I reverted the protected methods back to private.